src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -271,6 +279,23 @@ export default class FilterOptionsComponent extends Component {
     if (this.config.storeOnChange) {
       this.apply(this.config.isDynamic);
     }
+
+    if (!this.config.isDynamic) {
+      this.core.globalStorage.on('update', this.name, (data) => {
```

Comment:
I don't have the previous back nav changes down by heart, how exactly does hitting the back button cause this globalStorage module to update?

Reply:
On back nav in `answers-umd.js` resetListener to persistentStorage, we `globalStorage.setAll()` with all the URL query params (which would include this filter name since filter params are added automatically to the URL)